features/lorawan/lorastack/mac/LoRaMacCommand.cpp

```diff
@@ -322,22 +322,15 @@ lorawan_status_t LoRaMacCommand::process_mac_commands(const uint8_t *payload, ui
                 secs |= (uint32_t) payload[mac_index++] << 8;
                 secs |= (uint32_t) payload[mac_index++] << 16;
                 secs |= (uint32_t) payload[mac_index++] << 24;
-                uint32_t millis = ((uint32_t) payload[mac_index++] * 1000) >> 8;
 
                 // round milliseconds to closest second
-                if (millis > 499) {
+                if (payload[mac_index++] >= 0x80) {
```

Comment:
Now that you're not just setting the RTC with this, wouldn't it make sense to keep the full resolution - work in milliseconds?